### PR TITLE
Iss hipcms 666 implement role check using authv3

### PR DESCRIPTION
**Pull request by nagaraj1993**

### CODE REVIEW

File: HiP-FeatureToggle/Controllers/FeatureGroupsController.cs [hunk 1]
```diff
@@ -2,28 +2,28 @@
...
 using System.Linq;
+using Microsoft.AspNetCore.Authorization;
+using PaderbornUniversity.SILab.Hip.FeatureToggle.Data;
 
 namespace PaderbornUniversity.SILab.Hip.FeatureToggle.Controllers
 {
     /// <summary>
     /// Provides methods to add/remove feature groups and to assign users to these groups.
     /// </summary>
+    [Authorize]
```
Comment (janis-kra) [inline HiP-FeatureToggle/Controllers/FeatureGroupsController.cs]: I think we should set this to `Authorize("read:featuretoggle")`, so all calls to this API require the `"read:featuretoggle"` scope. Additionally, for all calls that require write privileges, the `"write:featuretoggle"` scope should be required (see below)

File: HiP-FeatureToggle/Controllers/FeatureGroupsController.cs [hunk 2]
```diff
@@ -72,7 +72,7 @@ public IActionResult GetById(int groupId)
         [ProducesResponseType(422)]
```
Comment (janis-kra) [inline HiP-FeatureToggle/Controllers/FeatureGroupsController.cs:74]: Add `[Authorize("write:featuretoggle")]` here, s.t. ppl without write privileges cannot call this method. Same for all other POST, PUT or DELETE calls in all the contorllers.

File: HiP-FeatureToggle/Controllers/FeaturesController.cs [hunk 3]
```diff
@@ -3,29 +3,28 @@
...
 using System.Collections.Generic;
 using System.Linq;
+using PaderbornUniversity.SILab.Hip.FeatureToggle.Data;
 
 namespace PaderbornUniversity.SILab.Hip.FeatureToggle.Controllers
 {
     /// <summary>
     /// Provides methods to add/modify/remove features.
     /// </summary>
+    [Authorize]
```
Comment (janis-kra) [inline HiP-FeatureToggle/Controllers/FeaturesController.cs]: See the comment in the other controller

File: HiP-FeatureToggle/Startup.cs [hunk 4]
```diff
@@ -70,7 +80,7 @@ public void ConfigureServices(IServiceCollection services)
             IHostingEnvironment env,
             ILoggerFactory loggerFactory,
             IOptions<AppConfig> appConfig,
-            ToggleDbContext dbContext)
+            ToggleDbContext dbContext, IOptions<AuthConfig> authConfig)
```
Comment (janis-kra) [inline HiP-FeatureToggle/Startup.cs]: missing a newline here

**Review by janis-kra — CHANGES_REQUESTED** [review]
Requsted a few changes that I did not think of before. Apart from that, good job. When you applied the changes just have someone else do a quick review. Everything else is fine I already tested it. Or just wait until I return ;)

If you don't have the time to implement the changes because you're going on vacation, maybe ping Sven and ask if he can do it. Otherwise I'll do it next week when I return.

**Comment by nagaraj1993** [comment]
Resharper check in TFS fails and hence build is getting partially
succeeded. I don't know why this error suddenly came so wierd which did
not come up in the previous commits. So I added the check which resharper suggested too but still the problem exists. Please check details. Keeping this open till @janis-kra is back from Vacation.

File: HiP-FeatureToggle/Startup.cs [hunk 5]
```diff
@@ -40,6 +40,7 @@ public void ConfigureServices(IServiceCollection services)
             services.Configure<AppConfig>(Configuration);
             services.Configure<AuthConfig>(Configuration.GetSection("Auth"));
 
+            //Adding authorization
```
Comment (janis-kra) [inline HiP-FeatureToggle/Startup.cs]: by the way, you can trigger new builds via TFS

**Comment by janis-kra** [comment]
The reason your NullReference check did not get rid of the R# error is that a `catch` block only catches errors that occur within the `try` block. The potential error that R# reported would have occured within another `catch` block and would thus not have been caught by your check

**Review by janis-kra — APPROVED** [review]

**Pull request merged**